multiplayer_server/fns/promote_to_moderator.php

```diff
@@ -114,13 +131,24 @@ function promote_mod($port, $name, $type, $admin, $promoted_player) {
 			}
 			
 			//action log
-			$admin_name = $admin->name;
-			$promoted_name = $promoted_player->name;
-			
+			$safe_admin_name = addslashes($admin->name);
+			$safe_promoted_name = addslashes($name);
 			$ip = $admin->ip;
 			
+			//make pretty server names
+			$servers = json_decode(file_get_contents('https://pr2hub.com/files/server_status_2.txt'));
+			$server_count = count($servers->servers);
+			
+			foreach (range(0,$server_count) as $server_id) {
+				$server_port = $servers->servers[$server_id]->port;
+				if ($port == $server_port) {
+					$server_name = $servers->servers[$server_id]->server_name;
+					break;
+				}
+			}
+			
 			// log action in action log
-			$db->call('mod_action_insert', array($admin->user_id, "$admin_name promoted $promoted_name to a $type moderator from $ip on port $port", $admin->user_id, $ip));
```

Comment:
`$db->call` escapes all of the values passed to it. Doing it manually here will double escape the values. It doesn't matter much here, but if an admin had the name bo'b, they would come out of the db with the name bo\'b